src/pages/ReportDetailsPage.tsx

```diff
@@ -633,7 +636,13 @@ function ReportDetailsPage({policies, report, session, personalDetails}: ReportD
                     isVisible={isDeleteModalVisible}
                     onConfirm={deleteTransaction}
                     onCancel={() => setIsDeleteModalVisible(false)}
-                    onModalHide={() => ReportUtils.navigateBackAfterDeleteTransaction(navigateBackToAfterDelete.current)}
+                    onModalHide={() => {
+                        if (isTransactionDeleted.current && !navigateBackToAfterDelete.current) {
```

Comment:
I am not sure why we need to introduce `isTransactionDeleted.current`. Wouldn't this work with just the support of `navigateBackToAfterDelete.current`? Or is there a use case which will fail if we don't use `isTransactionDeleted.current`?

Reply:
> Wouldn't this work with just the support of navigateBackToAfterDelete.current?

Sorry, I don't quite understand this. Do you mean to just check `if (!navigateBackToAfterDelete.current) {`?

Reply:
> > Wouldn't this work with just the support of navigateBackToAfterDelete.current?
> 
> Sorry, I don't quite understand this. Do you mean to just check `if (!navigateBackToAfterDelete.current) {`?

Yes. That's correct. Looks like I needed to make this a little more clear in communication. I am curious to know a use case that would fail if we just check `if (!navigateBackToAfterDelete.current) {`.

Reply:
It will close the page when we close the modal without deleting the transaction (for example pressing the backdrop/overlay).

Reply:
> It will close the page when we close the modal without deleting the transaction (for example pressing the backdrop/overlay).

Ah! Ok. That makes sense. Since the purpose of `isTransactionDeleted` is not very obvious due to the presence of `navigateBackToAfterDelete`, can you please add relevant comments to the code to reflect the purpose of `isTransactionDeleted`? I think this will help someone who looks at this code later.

Reply:
Done